### PR TITLE
add endpoint and task for listing all course enrollments in a program

### PR DESCRIPTION
**Pull request by jansenk**
add endpoint and task for listing all course enrollments in a program

### CODE REVIEW

File: registrar/apps/enrollments/tasks.py [hunk 1]
```diff
@@ -113,6 +113,49 @@ def list_course_run_enrollments(
...
+def list_all_course_run_enrollments(self, job_id, user_id, file_format, program_key):
+    """
+    A user task for retrieving all course enrollments within a given program from LMS.
+    """
+    program = _get_program(job_id, program_key)
+    if not program:
+        return None
+
+    results = []
+    for course_run in program.discovery_program.course_runs:
```
Comment (schenedx) [inline registrar/apps/enrollments/tasks.py:126]: I think you need to get the list of course_runs from curriculum, not from this property. See https://github.com/edx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L1706

File: registrar/apps/enrollments/tests/test_tasks.py [hunk 2]
```diff
@@ -194,6 +195,51 @@ def spawn_task(self, program_key=None, **kwargs):
...
+        super().setUpTestData()
+        for enrollment in cls.enrollment_data:
+            enrollment['course_key'] = cls.external_course_key
+
+    def setUp(self):
+        super().setUp()
+        program_patcher = mock.patch.object(DiscoveryProgram, 'get')
+        self.mocked_discovery_program = program_patcher.start()
+        self.mocked_discovery_program.return_value = DiscoveryProgram(
+            course_runs=[
```
Comment (schenedx) [inline registrar/apps/enrollments/tests/test_tasks.py:220]: You need to setup the course_runs in tests through Curriculum

**Review by schenedx — CHANGES_REQUESTED** [review]
See my inline feedback

File: registrar/apps/enrollments/tasks.py (hunk 1 above)
Comment (jansenk) [reply registrar/apps/enrollments/tasks.py:126]: `discovery_program.course_runs` is that. it's a flattened list of all course runs in the active curriculum

https://github.com/edx/registrar/blob/41a378f1b527d67f7877f9c34f903fd37b508e75/registrar/apps/enrollments/data.py#L137

Comment (schenedx) [reply registrar/apps/enrollments/tasks.py:126]: Oh... We are in registrar context, not in Publisher. My bad ✋ ➡️ 🤦‍♂

**Review by schenedx — APPROVED** [review]
👍

File: registrar/apps/api/v1/views.py [hunk 3]
```diff
@@ -484,3 +485,40 @@ class CourseRunEnrollmentUploadView(ProgramSpecificViewMixin, EnrollmentUploadVi
...
+     * 403: User lacks enrollment read access to organization of specified program
+     * 404: Program was not found.
+
+    Example Response:
+    {
+        "job_id": "3b985cec-dcf4-4d38-9498-8545ebcf5d0f",
+        "job_url": "http://localhost/api/[version]/jobs/3b985cec-dcf4-4d38-9498-8545ebcf5d0f"
+    }
+
+    Path: /api/[version]/programs/{program_key}/course_enrollments
```
Comment (jansenk) [inline registrar/apps/api/v1/views.py:507]: Changed this to [version] to match #147

**Pull request merged**